spec/unit/rest/client_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'spec_helper'
+
+require 'puppet/rest/client'
+require 'puppet/rest/route'
+
+describe Puppet::Rest::Client do
+  context "when creating a new client" do
+    let(:route) { Puppet::Rest::Route.new(api: "/fake_api/v1/",
+                                          default_server: "myserver.com",
+                                          default_port: 555) }
+    let(:ssl_store) { mock('store') }
+    let(:ssl_config) { stub_everything('ssl config') }
+    let(:http) { stub_everything('http', :ssl_config => ssl_config) }
+
+    before(:each) do
+      route.stubs(:select_server_and_port).returns(["myserver.com", 555])
+    end
+
+    it "configures a base URL based on the provided route" do
+      url = "https://myserver.com:555/fake_api/v1/"
+      http.expects(:base_url=).with() do |arg|
+        arg.to_s == url
+      end
+      client = Puppet::Rest::Client.new(route,
+                                        client: http,
+                                        ssl_store: ssl_store)
+    end
+
+    it "initializes itself with basic defaults" do
+      HTTPClient.expects(:new).returns(http)
+      OpenSSL::X509::Store.expects(:new).returns(ssl_store)
+      # Configure connection with HTTP settings
+      Puppet.expects(:[]).with(:http_user_agent)
+      Puppet.expects(:[]).with(:http_read_timeout).returns(120)
+      Puppet.expects(:[]).with(:http_connect_timeout).returns(10)
+      Puppet.expects(:[]).with(:http_debug).returns(true)
```

Comment:
Puppet's spec helper resets settings across tests, so you can just set the properties directly: `Puppet[:http_user_agent] = 'whatever'`